File: Array/1.php
<?php

class Solution
{

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum(array $nums, int $target)
    {
        $result = [];

        foreach ($nums as $key => $num) {
            $diff = $target - $num; // 9 - 2 => 7 |||| 9 - 7 = 2

            if (isset($result[$diff])) {
                return [$result[$diff], $key];
            }

            $result[$num] = $key; // [7: 0] |||| [2: 1]
        }

        return null;
    }
}

$solution = new Solution();
print_r($solution->twoSum([2, 7, 11, 15], 9));
//print_r($solution->twoSum([3, 2, 4], 6));

File: problems/Array/1.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        result: dict = {}

        for key, num in enumerate(nums):
            request = target - num

            if request in result:
                return [result[request], key]

            result[num] = key

        return []


if __name__ == '__main__':
    print(Solution().twoSum([2, 7, 11, 15], 9))
